### PR TITLE
Add confirmation popup for deletion

### PR DESCRIPTION
**Pull request by jonas-tai**
A new popup that asks the user for confirmation is shown, before an attribution is deleted. The confirmation popup is not shown, if the attribution was preselected.

Signed-off-by: Jonas Tai <jonas.tai@tngtech.com>

### CODE REVIEW

File: src/Frontend/Components/ConfirmationPopup/__tests__/ConfirmPopup.test.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// SPDX-FileCopyrightText: Facebook, Inc. and its affiliates
```
Comment (benedikt-richter) [inline src/Frontend/Components/ConfirmationPopup/__tests__/ConfirmPopup.test.tsx:1]: the naming of this file is inconsistent with the naming of the respective component

File: src/Frontend/Components/ConfirmDeletionPopup/__tests__/ConfirmDeletionPopup.test.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+import { renderComponentWithStore } from '../../../test-helpers/render-component-with-store';
+import { screen } from '@testing-library/react';
+import React from 'react';
+import { ConfirmDeletionPopup } from '../ConfirmDeletionPopup';
+
+describe('The ConfirmDeletionPopup', () => {
+  test('renders', () => {
+    const expectedContent = 'Do you really want to delete this attribution?';
```
Comment (benedikt-richter) [inline src/Frontend/Components/ConfirmDeletionPopup/__tests__/ConfirmDeletionPopup.test.tsx:13]: Could be `'Do you really want to delete this attribution for the current file?'`

File: src/Frontend/Components/ConfirmDeletionPopup/ConfirmDeletionPopup.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      );
+    } else {
+      dispatch(deleteAttributionForAllAndSave(selectedAttributionId));
+    }
+  }
+
+  return (
+    <ConfirmationPopup
+      onConfirmation={deleteAttributionForResource}
+      content={'Do you really want to delete this attribution?'}
```
Comment (benedikt-richter) [inline src/Frontend/Components/ConfirmDeletionPopup/ConfirmDeletionPopup.tsx:46]: Could be `'Do you really want to delete this attribution for the current file?'`

File: src/Frontend/Components/ConfirmDeletionPopup/ConfirmDeletionPopup.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  const view = useSelector(getSelectedView);
+  const selectedResourceId = useSelector(getSelectedResourceId);
+  const selectedAttributionId = useSelector(getSelectedAttributionId);
+  const attributionIdOfSelectedPackageInManualPanel: string | null =
+    useSelector(getAttributionIdOfDisplayedPackageInManualPanel);
+
+  const dispatch = useDispatch();
+
+  function deleteAttributionForResource(): void {
+    if (view == View.Audit && attributionIdOfSelectedPackageInManualPanel) {
```
Comment (benedikt-richter) [inline src/Frontend/Components/ConfirmDeletionPopup/ConfirmDeletionPopup.tsx:31]: akways use `===`

File: src/Frontend/Components/ConfirmDeletionPopup/ConfirmDeletionPopup.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      dispatch(
+        deleteAttributionAndSave(
+          selectedResourceId,
+          attributionIdOfSelectedPackageInManualPanel
+        )
+      );
+    } else {
+      dispatch(deleteAttributionForAllAndSave(selectedAttributionId));
+    }
+  }
```
Comment (benedikt-richter) [inline src/Frontend/Components/ConfirmDeletionPopup/ConfirmDeletionPopup.tsx:41]: The logic is broken:
If I select an attribution in audit view in attributions in folder content and press delete. This attribution is deleted for all

**Review by benedikt-richter — CHANGES_REQUESTED** [review]

File: src/Frontend/Components/ConfirmDeletionPopup/ConfirmDeletionPopup.tsx (hunk 4 above)
Comment (nicarl) [reply src/Frontend/Components/ConfirmDeletionPopup/ConfirmDeletionPopup.tsx:31]: This check for the view seems wrong as Benedikt said. I would rather expect a check if the attribution is associated to several files. I think we have infrastructure in the selector for this.

**Review by benedikt-richter — APPROVED** [review]

**Pull request merged**